packages/react-scripts/scripts/utils/verifyPackageTree.js

```diff
@@ -13,22 +13,27 @@ const fs = require('fs');
 const semver = require('semver');
 const path = require('path');
 
+const isESLintPluginEnabled = process.env.DISABLE_ESLINT_PLUGIN !== 'true';
+
 // We assume that having wrong versions of these
 // in the tree will likely break your setup.
 // This is a relatively low-effort way to find common issues.
 function verifyPackageTree() {
-  const depsToCheck = [
+  let depsToCheck = [
     // These are packages most likely to break in practice.
     // See https://github.com/facebook/create-react-app/issues/1795 for reasons why.
     // I have not included Babel here because plugins typically don't import Babel (so it's not affected).
-    'babel-eslint',
     'babel-jest',
     'babel-loader',
-    'eslint',
     'jest',
     'webpack',
     'webpack-dev-server',
   ];
+
+  if (isESLintPluginEnabled) {
+    depsToCheck = [...depsToCheck, 'babel-eslint', 'eslint'];
```

Comment:
You might as well just push `eslint` and `babel-eslint` into the array and then you can avoid spreading and reassigning the previous value. That's how I would do it anyway, but I don't feel strongly about this.

Reply:
I don't disagree, it's a stylistic thing. _Most_ of my the projects I work on are written without mutation... and I just do it by default now.

Reply:
This was updated to use `filter(Boolean)` instead, as we do in other places.